src/main/java/com/github/princesslana/eriscasper/rest/Routes.java

```diff
@@ -47,15 +44,13 @@ public Routes(BotToken token, OkHttpClient client, ObjectMapper jackson) {
   }
 
   public <I, O> Single<O> execute(Route<I, O> route, I data) {
-    Function<Optional<RequestBody>, Request> buildRequest =
-        body ->
-            route.newRequestBuilder(data).header("Authorization", "Bot " + token.unwrap()).build();
-
-    return Maybes.fromNullable(data)
-        .map(d -> RequestBody.create(MEDIA_TYPE_JSON, jackson.writeValueAsString(data)))
-        .map(Optional::of)
-        .toSingle(Optional.empty())
-        .map(buildRequest)
+    Function<Request.Builder, Request.Builder> addAuthorizationHeader =
```

Comment:
Should we consider making this a constant? or at least a private final field in the class. Also, the naming is kinda weird, verbs are mainly used as methods. A more conventional naming here would be something like `addAuthorizationHeaderFunction`. Just if we want to follow standard naming conventions. Not really big just a nit.